src/rules/utils.ts

```diff
@@ -758,3 +758,32 @@ export const scopeHasLocalReference = (
     !references.unresolved.has(referenceName)
   );
 };
+
+/**
+ * Checks if node names returned by getNodeName matches any of the given star patterns
+ * Pattern examples:
+ *   request.*.expect
+ *   request.**.expect
+ *   request.**.expect*
+ */
+
+export function nodeNameMatch(
+  nodeName: string,
+  ...patterns: readonly string[]
+): boolean {
+  for (const p of patterns) {
+    if (
+      new RegExp(
```

Comment:
[This should use the `u` flag](https://eslint.org/docs/rules/require-unicode-regexp).

(I'll add this to our eslint config soon)